### PR TITLE
Update ieaffocus.cpp

### PR DESCRIPTION
**Pull request by joe13905179063**
add iOptron iAFS focuser support

### CODE REVIEW

**Comment by joe13905179063** [comment]
add iEFW indi driver

File: drivers/filter_wheel/ioptron_wheel.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+        bool getiEFWID();
+	bool getiEFWInfo();
+        bool getiEFWfirmwareInfo();
+	void initOffset();
+  // Calibration offset
+        bool getOffset(int filter);
+        bool setOffset(int filter, int shift);
+        ISwitch HomeS[1];
+        ISwitchVectorProperty HomeSP;
```
Comment (knro) [inline drivers/filter_wheel/ioptron_wheel.h:57]: Please use INDI new style properties. i.e.

INDI::PropertyText
INDI::PropertySwitch

and so forth

**Comment by joe13905179063** [comment]
Jasem, updated using INDI new style properties. Thanks

File: drivers/filter_wheel/ioptron_wheel.cpp [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+
+const char *iEFW::getDefaultName()
+{
+    return "iOptron Wheel";
+}
+
+bool iEFW::initProperties()
+{
+    INDI::FilterWheel::initProperties();
+    serialConnection->setDefaultPort("/dev/ttyUSB3");
```
Comment (knro) [inline drivers/filter_wheel/ioptron_wheel.cpp:73]: I don't think it should be set to this port. Please remove this line as this is a generic port.

File: drivers/filter_wheel/ioptron_wheel.cpp [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+        tty_error_msg(rc, errstr, MAXRBUF);
+        DEBUGF(INDI::Logger::DBG_ERROR, "Init send iEFW deviceinfo  error: %s.", errstr);
+        return false;
+    };
+
+    if ( (rc = tty_read_section(PortFD, resp, '#', iEFW_TIMEOUT * 2, &nbytes_read)) != TTY_OK)
+    {
+        char errstr[MAXRBUF] = {0};
+        tty_error_msg(rc, errstr, MAXRBUF);
+        DEBUGF(INDI::Logger::DBG_ERROR, "Init read iEFW deviceinfo error: %s.", errstr);
```
Comment (knro) [inline drivers/filter_wheel/ioptron_wheel.cpp:155]: LOGF_ERROR instead of DEBUGF

File: drivers/filter_wheel/ioptron_wheel.cpp [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+         WheelIDTP.setState(IPS_OK);
+         WheelIDTP[0].setText("iEFW-18");
+         WheelIDTP.apply();
+
+        };
+	return true;
+    }
+    else
+    {
+        DEBUGF(INDI::Logger::DBG_ERROR, "iEFW getinfo Response: %s", resp);
```
Comment (knro) [inline drivers/filter_wheel/ioptron_wheel.cpp:182]: LOGF_ERROR instead of DEBUGF

File: drivers/filter_wheel/ioptron_wheel.cpp [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+{
+    int nbytes_written = 0, nbytes_read = 0, rc = -1;
+    char errstr[MAXRBUF];
+    char resp[16] = {0};
+    char iefwfirminfo[16] = {0};
+    tcflush(PortFD, TCIOFLUSH);
+    if ( (rc = tty_write(PortFD, ":FW1#", 5, &nbytes_written)) != TTY_OK)
+    {
+        tty_error_msg(rc, errstr, MAXRBUF);
+        DEBUGF(INDI::Logger::DBG_ERROR, "get iEFW FiremwareInfo error: %s.", errstr);
```
Comment (knro) [inline drivers/filter_wheel/ioptron_wheel.cpp:201]: LOGF_ERROR instead of DEBUGF

File: drivers/filter_wheel/ioptron_wheel.cpp [hunk 6]
```diff
@@ -0,0 +1,331 @@
...
+        return true;
+}
+
+/////////////////////////////////////////////////////////////////////////////
+bool iEFW::getOffset(int filter)
+{
+        return true;
+}
+*/
+/////////////////////////////////////////////////////////////////////////////
```
Comment (knro) [inline drivers/filter_wheel/ioptron_wheel.cpp:283]: remove unused code

**Comment by joe13905179063** [comment]
ok.

File: drivers/filter_wheel/ioptron_wheel.cpp [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+
+bool iEFW::ISNewSwitch(const char *dev, const char *name, ISState *states, char *names[], int n)
+{
+    if (dev != nullptr && strcmp(dev, getDeviceName()) == 0)
+    {
+        if (HomeSP.isNameMatch(name))
+        {
+            if (getiEFWID())
+            {
+                LOG_INFO("Filter getid.");
```
Comment (knro) [inline drivers/filter_wheel/ioptron_wheel.cpp:115]: Perhaps a better message should displayed to the user?

LOG_INFO("Filter is at home position.");

for example?

**Comment by joe13905179063** [comment]
thanks

**Pull request merged**